pkg/controller/rhmiconfig/helpers/update_status.go

```diff
@@ -0,0 +1,101 @@
+package helpers
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const WINDOW = 6
+
+func UpdateStatus(ctx context.Context, client k8sclient.Client, config *integreatlyv1alpha1.RHMIConfig, targetVersion string) error {
+	// Calculate the next maintenance window based on the maintenance schedule
+	if config.Spec.Maintenance.ApplyFrom != "" {
+		mtStart, _, err := getWeeklyWindowFromNow(config.Spec.Maintenance.ApplyFrom, time.Hour*WINDOW)
+		if err != nil {
+			return err
+		}
+
+		config.Status.Maintenance.ApplyFrom = mtStart.Format("2-1-2006 15:04")
+		config.Status.Maintenance.Duration = strconv.Itoa(WINDOW) + "hrs"
+	}
+
+	config.Status.UpgradeAvailable.TargetVersion = targetVersion
+
+	client.Status().Update(ctx, config)
+
+	// Calculate the upgrade schedule based on the spec:
+	// We can assume there's no error parsing the value as it was validated
+	notBeforeDays := *config.Spec.Upgrade.NotBeforeDays
+	waitForMaintenance := *config.Spec.Upgrade.WaitForMaintenance
+
+	upgradeSchedule := config.Status.UpgradeAvailable.AvailableAt.
+		Add(daysDuration(notBeforeDays))
+
+	if waitForMaintenance {
+		var err error
+		upgradeSchedule, _, err = getWeeklyWindow(upgradeSchedule, config.Spec.Maintenance.ApplyFrom, time.Hour*WINDOW)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Update the upgrade status
+	config.Status.Upgrade = integreatlyv1alpha1.RHMIConfigStatusUpgrade{
+		Scheduled: &integreatlyv1alpha1.UpgradeSchedule{
+			For: upgradeSchedule.Format(integreatlyv1alpha1.DateFormat),
+		},
+	}
+
+	return client.Status().Update(ctx, config)
```

Comment:
```suggestion
	return nil
```

As mentioned in the suggestion above, the `client.Status().Update()` call can be performed afterwards, so there's no need to call it in the `UpdateStatus` function

Reply:
Hey Sergio, thank you for very valuable comments. When these changes were applied it seems that the `upgrade` from the RHMIConfig is empty, and in my understanding, because we are not making API call to upgrade the RHMIConfig CR inside the updateStatus function anymore and returning a nil value, we are returning an empty upgrade value. I think the update.Status() should be somewhere within the UpdateStatus or should we return a config with this value updated and then do client.Update in the RHMIConfig reconciler?

Reply:
Notice that the return type of the UpdateStatus function is `error`, so by returning `nil` we're confirming that the function was successful. We don't need to return the config because the function mutates the one that's passed as a parameter as a reference